package.json

```
@@ -50,32 +58,127 @@
         "path": "./syntaxes/markdown.tmLanguage.json"
       }
     ],
-    "themes": [
+    "commands": [
       {
-        "label": "D2 Light",
-        "uiTheme": "vs",
-        "path": "./themes/light-color-theme.json"
-      },
+        "command": "d2-viewer.ShowPreviewWindow",
+        "title": "Show d2 Preview Window",
+        "category": "d2"
+      }
+    ],
+    "keybindings": [
       {
-        "label": "D2 Dark",
-        "uiTheme": "vs-dark",
-        "path": "./themes/dark-color-theme.json"
+        "command": "d2-viewer.ShowPreviewWindow",
+        "key": "ctrl+shift+d",
+        "mac": "shift+cmd+d"
       }
-    ]
+    ],
+    "configuration": {
+      "type": "object",
+      "title": "d2 Viewer Configuration",
+      "properties": {
+        "d2-viewer.autoUpdate": {
+          "type": "boolean",
+          "default": true,
+          "description": "Automatically update the preview."
+        },
+        "d2-viewer.updateTimer": {
+          "type": "number",
+          "default": 1500,
+          "description": "Specifies the amount of time to wait between keystrokes before automatically updating the preview. (1000 = 1000ms = 1s)"
+        },
+        "d2-viewer.updateOnSave": {
+          "type": "boolean",
+          "default": false,
+          "description": "Automatically update on saving the d2 file.  Turn off to avoid excess updates when Auto Save is enabled."
+        }
+      }
+    },
+    "menus": {
+      "commandPalette": [
+        {
+          "command": "d2-viewer.ShowPreviewWindow",
```

```suggestion
          "command": "D2.ShowPreviewWindow",
```
maybe just D2 for these?

---

I was afraid this would cause name collisions, but it's scoped to the extension.  Fixed in PR #16.

---

this PR?

---

The next PR.  This one is getting a little out of hand.

---

Updated this PR